src/BatchProcessor.ts

```diff
@@ -1,19 +1,245 @@
 /* Copyright © 2024 Seneca Project Contributors, MIT License. */
 
+import Inks from 'inks'
+import { Gubu } from 'gubu'
+
+const ALL = '*'
+const Types: Record<string, number> = {
+  'Number': 1,
+  'Boolean': 1,
+  'String': 1,
+  'Array': 1,
+  'Object': 1
+}
 
 type BatchProcessorOptionsFull = {
   debug: boolean
+  send: any
+  where: any
 }
 
 export type BatchProcessorOptions = Partial<BatchProcessorOptionsFull>
 
 
 // FEATURE: subsets of keys by dot separators.
+class Match {
+  all: boolean
+  patrun: any
+  
+  constructor(patrun: any, all: any = null) {
+    this.patrun = patrun
+    this.all = all
+  }
+  
+  set_all(all: boolean) {
+    this.all = all
+  }
+  
+  find(pattern: any) {
+    return this.patrun.find(pattern)
+  }
+  
+  add_pattern(pattern: any, item: any) {
+    return this.patrun.add(pattern, item)
+  }
+  
+}
+
+class Utility {
```

Comment:
don't use classes as function containers - classes should be only for things that are instantiated